### ISSUE TITLE
[1.18.2] Copper tools able to mine obsidian

### ISSUE THREAD

**Issue by noelforte** [kwpugh/EmeraldTools#23]
It seems like copper tools are able to break obsidian blocks, is this a bug? Since copper is less durable than iron and iron tools can't mine obsidian (if you break obsidian with iron it drops nothing), it seems to not make sense that obsidian is more readily available if you have copper tools than it would otherwise be if you were only able to acquire it when using either obsidian, ruby, emerald, diamond etc. Love the mod so far!

**Comment by kwpugh** [comment]
Copper should not break obsidian.  Which version of the mod are you using.

**Comment by noelforte** [comment]
EmeraldTools-Fabric-1.18.2-1.3.43.jar, running on a server using 1.18.2, same version mirrored on server and client

**Comment by kwpugh** [comment]
ok, must be something more going on here.  I tried on single player and on my server.  copper pickaxe could not break obsidian or crying obsidian.  Copper excavator and hammer could not break adjacent obsidian either.

Can you explain more details about what you are seeing when it occurs?

**Comment by kwpugh** [comment]
Oh, I see.  if you beat on the obsidian until it breaks it drops the block, which it should not.

**Comment by kwpugh** [comment]
Ok, the problem is that I some how set the default mining level of the copper tool material to 3 (diamond) instead of 2 (iron).

I will fix the default in next update, but you can change on your side in the config:

	"copperMiningLevel": 2

Please confirm that fixes it for you.

**Comment by kwpugh** [comment]
Same with steel tool material:

	"steelMiningLevel": 2,

**Comment by noelforte** [comment]
Updated the config, that doesn't appear to change the behavior

**Comment by kwpugh** [comment]
Both configs?

Client and Server.

**Comment by noelforte** [comment]
Yep, with client and server configurations changed I'm still able to get obsidian blocks with copper tools. Is it by design that the client and server can run with different settings? Because if so, that seems like an issue in itself if players can override the server config by editing their client config.

**Comment by kwpugh** [comment]
usually no.  The mining level is set by the server config.  I get no obsidian drops with the copper pickaxe, hammer, or paxel.  Did you restart the server after the config change or just a reload?

**Comment by kwpugh** [comment]
Also, what other mods are you running on the server?

**Comment by noelforte** [comment]
Ok I think we're getting somewhere. Just tested with the Paxel and Pickaxe and can confirm, setting copper `"copperMiningLevel": 2" achieves intended behavior, however the hammer continues to be able to generate obsidian drops. I've been playing on survival, are those stats cached on a per-item basis or are they applied globally?

Mod list is as follows:

```
architectury          4.3.53
cloth-config          6.2.57-fabric
consistency_plus      0.5.0+1.18.2-fabric
EmeraldTools-fabric   1.18.2-1.3.43
fabric-api            0.51.1+1.18.2
Incendium             5.0.3
lithium               18.2-0.7.9
sodium                18.2-0.4.1+build.15
starlight             1.0.2+fabric.89b8d9f
Terralith             2.2.1a
```

**Comment by kwpugh** [comment]
I don't understand what you are asking:  "are those stats cached on a per-item basis or are they applied globally?"

**Comment by noelforte** [comment]
What I meant by that was "if I were to acquire a different copper hammer would that hammer have the lower mining level than the one that I already had" in other words, are the stats cached for each individual item? I answered my question though because even with destroying and creating a new instance of a copper hammer, the copper hammer is able to get obsidian drops.

As I said above, I've tested the paxel and pickaxe and those appear to be working correctly now, I can't get obsidian drops with those, just the hammer.

**Comment by kwpugh** [comment]
oh, I see.  So, no, material values are loaded at startup and applied to ALL items of that material, no matter what they were before.

NOTE:  I have covid 19 and my brain is a bit foggy as a result.

I thought I tested the the hammer, but I didn't.  I just tested the hammer and it breaks obsidian with drops.  I looked into my code and I can see why.  I do a manual check to see if the block state is suitable to break by a diamond pickaxe.  

I will look again in the morning to see how I can fix that.
